src/BSON/ObjectID.c

```diff
@@ -75,28 +77,33 @@ static bool php_phongo_objectid_init_from_hex_string(php_phongo_objectid_t *inte
 		return true;
 	}
 
+	phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Invalid BSON ID provided");
```

Comment:
This is the same error message used in HHVM (see: [here](https://github.com/mongodb/mongo-hhvm-driver/blob/73bc3daa7691184a6e091585ad42c01cb965fe41/src/MongoDB/BSON/ObjectID.cpp#L42)), but I think there's room for improvement. We should at least show the offending value.

Let's discuss and create a new ticket if needed.